tests/typerel/tphantom_type_as_base.nim

```diff
@@ -72,4 +72,25 @@ block with_ptr_object:
 
   proc p(x: ptr Sub): int = 3
   # the more precise overload matches:
-  doAssert p(addr obj) == 3
\ No newline at end of file
+  doAssert p(addr obj) == 3
+
+block with_intermediate_non_object_base:
+  # test the case where an intermediate base type is a ``ref`` or
+  # ``ptr`` while formal type is not
+  type
+    Root[T] = object of RootObj
+
+    First   = ref object of Root[int]
+    Second  = object of First
```

Comment:
no change for this PR. musing outloud really, but I do wonder about this feature... maybe it's ok under the "what else could it be?" although those sorts of things preclude future extension of the language.